terraform/cluster/aws/main.tf

```diff
@@ -102,7 +102,7 @@ resource "aws_eks_node_group" "cluster" {
   instance_types  = split(",", random_id.node_group.keepers.node_type)
   node_group_name = "${var.name}-${local.availability_zones[count.index]}-${count.index}${random_id.node_group.hex}"
   node_role_arn   = random_id.node_group.keepers.role_arn
-  subnet_ids      = [var.private ? aws_subnet.private[count.index].id : aws_subnet.public[count.index].id]
+  subnet_ids      = [var.private ? var.private_subnets_ids[count.index] : aws_subnet.public[count.index].id]
```

Comment:
I think will not be able to use the default subnet created by this module  anymore `aws_subnet.private[count.index].id` 
 can we use `local.private_subnets_ids[count.index]` instead and have this logic there?
 or add a double condition here like aws_subnet.private resource?